app/keepers/keys.go

```diff
@@ -0,0 +1,110 @@
+package keepers
+
+import (
+	"github.com/CosmWasm/wasmd/x/wasm"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	authzkeeper "github.com/cosmos/cosmos-sdk/x/authz/keeper"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	evidencetypes "github.com/cosmos/cosmos-sdk/x/evidence/types"
+	"github.com/cosmos/cosmos-sdk/x/feegrant"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/cosmos/cosmos-sdk/x/group"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/cosmos/cosmos-sdk/x/quarantine"
+	"github.com/cosmos/cosmos-sdk/x/sanction"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	icqtypes "github.com/cosmos/ibc-apps/modules/async-icq/v6/types"
+	icahosttypes "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/host/types"
+	ibctransfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	ibchost "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+
+	attributetypes "github.com/provenance-io/provenance/x/attribute/types"
+	"github.com/provenance-io/provenance/x/exchange"
+	"github.com/provenance-io/provenance/x/hold"
+	ibchookstypes "github.com/provenance-io/provenance/x/ibchooks/types"
+	ibcratelimit "github.com/provenance-io/provenance/x/ibcratelimit"
+	markertypes "github.com/provenance-io/provenance/x/marker/types"
+	metadatatypes "github.com/provenance-io/provenance/x/metadata/types"
+	msgfeestypes "github.com/provenance-io/provenance/x/msgfees/types"
+	nametypes "github.com/provenance-io/provenance/x/name/types"
+	oracletypes "github.com/provenance-io/provenance/x/oracle/types"
+	rewardtypes "github.com/provenance-io/provenance/x/reward/types"
+	triggertypes "github.com/provenance-io/provenance/x/trigger/types"
+
+	_ "github.com/provenance-io/provenance/client/docs/statik" // registers swagger-ui files with statik
+)
+
+func (appKeepers *AppKeepers) GenerateKeys() {
```

Comment:
There's some changes to key handling in v0.50: https://github.com/cosmos/cosmos-sdk/blob/v0.50.1/UPGRADING.md#module-wiring

We should probably wait to see what that looks like before overhauling this stuff.

Reply:
Yeah, I'll have to see how that plays out and integrate it into this.